src/Socket.hpp

```diff
@@ -275,16 +275,15 @@ class Socket {
 #endif
 		DEFAULT_MAX_CONN_QUEUE = 128
 	};
-	enum sock_type {
-	  BF_SOCK_DGRAM  = SOCK_DGRAM,
-		BF_SOCK_STREAM = SOCK_STREAM
-	};
 	// Manage an existing socket (usually one returned by Socket::accept())
 	// TODO: With C++11 this could return by value (moved), which would be nicer
 	inline static Socket* manage(int fd) { return new Socket(fd, ManageTag()); }
-	inline explicit       Socket(/*sock_type*/int type=SOCK_DGRAM)
-		: _fd(-1), _type((sock_type)type), _family(AF_UNSPEC),
-		  _mode(Socket::MODE_CLOSED) {}
+	inline explicit       Socket(int type=SOCK_DGRAM)
+		: _fd(-1), _type(type), _family(AF_UNSPEC),
+		  _mode(Socket::MODE_CLOSED) {
+		BF_ASSERT_EXCEPTION(type == SOCK_DGRAM || type == SOCK_STREAM,
+												BF_STATUS_INVALID_ARGUMENT);
```

Comment:
Hmm, indentation here is strange. I'm not accustomed to using tabs, but I let emacs retain tabs in files in this project for consistency, though I may be setting `tab-width` to 2. So then I think it inserted all tabs to get the 2nd line to line up after the parens, but doing that only works out if `tab-width` remains 2. Is "the right thing" to use tab char to line up with `BF_ASSERT` on first line, then spaces to get out to the parens? (This is why I don't use tabs, but I guess if I convince emacs to do the right thing it won't matter much.)

Reply:
Yeah, I run across strange indentation sometimes as well.  Maybe we should consider making to a move to all spaces, no tabs as some point.